### PR TITLE
Fix msg "'<' not supported between instances of 'str' and 'int'"

### PR DESCRIPTION
**Pull request by mistercrunch**
Druid returns NULL as 0, typed as int. This causes pandas to fail
when it tries to sort heterogeneous types.

### CODE REVIEW

**Review by betodealmeida — APPROVED** [review]
Should we change pydruid to do this automatically, or maybe convert the zero to `None`?

File: superset/connectors/druid/models.py [hunk 1]
```diff
@@ -1237,15 +1237,21 @@ def query(self, query_obj):
...
         cols = []
+        groupby = query_obj.get('groupby') or []
+        columns = query_obj.get('columns') or []
+        metrics = query_obj.get('metrics') or []
         if DTTM_ALIAS in df.columns:
             cols += [DTTM_ALIAS]
-        cols += query_obj.get('groupby') or []
-        cols += query_obj.get('columns') or []
-        cols += query_obj.get('metrics') or []
+        cols += groupby
```
Comment (xrmx) [inline superset/connectors/druid/models.py]: cols = [DTTM_ALIAS] + groupby + columns + metrics

may be a bit cheaper and clean

File: superset/connectors/druid/models.py [hunk 2]
```diff
@@ -1237,15 +1237,21 @@ def query(self, query_obj):
...
+        cols += columns
+        cols += metrics
 
         cols = [col for col in cols if col in df.columns]
         df = df[cols]
 
+        for col in groupby + columns:
```
Comment (xrmx) [inline superset/connectors/druid/models.py]: would it be possible for some of these columns to not be in df.columns given the list comprehension at line 1249?

**Comment by mistercrunch** [comment]
@betodealmeida I agree that pydruid should take care of that, but that change may not be backward compatible, so I'm not sure how to handle it.

I have to admit I'm confused to see the related bugs. I feel like it's a new behavior (new version of Druid? PyDruid? error in druid ingestion?) as I we should have had this problem before...

related? 
https://github.com/druid-io/druid/issues/4349

**Comment by mistercrunch** [comment]
Dug deeper and got to the root cause in https://github.com/apache/incubator-superset/pull/4358

**Pull request closed without merging**